components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/IdentityOathEventListener.java

```diff
@@ -474,4 +479,28 @@ private void removeClaimCacheEntry(String username, UserStoreManager userStoreMa
         }
         ClaimCache.getInstance().clearCacheEntry(cacheEntry.getClaimCacheKey());
     }
+
+    private void invokePostRevocationListeners(AccessTokenDO accessTokenDO) {
+
+        OAuthEventInterceptor oAuthEventInterceptorProxy = OAuthComponentServiceHolder.getInstance()
+                .getOAuthEventInterceptorProxy();
+        if (oAuthEventInterceptorProxy != null && oAuthEventInterceptorProxy.isEnabled()) {
+            try {
+                oAuthEventInterceptorProxy.onPostTokenRevocationBySystem(accessTokenDO, Collections.emptyMap());
+            } catch (IdentityOAuth2Exception e) {
+                log.error("Error occurred when invoking post token revoke listener ", e);
```

Comment:
Do we need log here? We can just throw here and log in the caller method.

Reply:
since we need to iterate over the tokens we just log not throw into upper level

Reply:
Samething should be applied for PreRevocation?

Reply:
Do you think, we need to skip revocation process through listeners?